documentation/writing_tasks.md

```diff
@@ -1077,3 +1077,453 @@ path.
 For more information on securely passing user input, see the blog post [Stop
 using backtick to run shell command in
 Ruby](https://www.hilman.io/blog/2016/01/stop-using-backtick-to-run-shell-command-in-ruby/).
+
+## Debugging tasks
+
+Because tasks are executed remotely, it can be challenging to debug tasks that
+are not working as expected. There are several ways that you can debug tasks,
+including using remote debugging libraries, using methods available in the
+task helper libraries, running the task locally as a script, and redirecting
+`stderr` to `stdout`.
+
+### Debug logs
+
+Typically, Bolt will only display task output that is sent to `stdout`.
```

Comment:
It might be useful to add that stderr will only be printed if nothing is sent to stdout. I get that it seems like TMI, but it's come up a few times just this week in channels and seems useful to document.

Reply:
I think in this specific case (using `--log-level debug`) it's clearer to set an expectation that task output will usually be `stdout` and that you can always see what was sent to `stderr` in debug mode.